app/src/main/java/org/systers/mentorship/view/fragments/TasksFragment.kt

```diff
@@ -43,49 +42,63 @@ class TasksFragment(private var mentorshipRelation: Relationship) : BaseFragment
         super.onActivityCreated(savedInstanceState)
 
         taskViewModel = ViewModelProviders.of(this).get(TasksViewModel::class.java)
-        taskViewModel.successful.observe(this, Observer {
+        taskViewModel.successfulGet.observe(this, Observer {
             successful ->
-            if (successful != null) {
-                if (successful) {
-                    if (taskViewModel.tasksList.isEmpty()) {
+            if (successful != null && successful) {
+                    if (taskViewModel.todoTasksList.isEmpty()) {
                         tvNoTask.visibility = View.VISIBLE
                         rvTasks.visibility = View.GONE
                     } else {
                         rvTasks.apply {
                             layoutManager = LinearLayoutManager(context)
-                            adapter = TasksAdapter(taskViewModel.tasksList, markTask)
+                            adapter = TasksAdapter(taskViewModel.todoTasksList, markTask)
                         }
                         tvNoTask.visibility = View.GONE
                     }
-                } else {
-                    view?.let {
-                        Snackbar.make(it, taskViewModel.message, Snackbar.LENGTH_LONG).show()
+
+                    if (taskViewModel.achievedTasksList.isEmpty()) {
```

Comment:
can you think of a way to refactor this : basically i have the List<Task> todoList and the List<Task> achievedTasksList and i have the same behaviour: when they're empty i display a message (but the content is different for both) and when they aren't empty i display the lis in question. I left it like this because it works, but potentially there's a way to make the code look more concise. would love some feed back on this thought